src/core/core.ts

```diff
@@ -128,13 +131,19 @@ class Core {
       this.config.httpConfig,
     );
 
-    const http = this.config.http;
+    // const http = this.config.http;
 
-    this.http = new http(httpConfig);
+    // TODO: Code Smell, http doesn't have a constructor.
+    // Should it be Axios instead of HttpAdapter?
+    // Everything was abstract which means this class wouldn't have anything implemented in it 
+    //  even if you were to instantiate it
+    // Also this.config.http most likely wouldn't have a Class stored in it which is also confusing
+    //  since it should be an instance
+    this.http = new AxiosAdapter(httpConfig);
```

Comment:
So this for sure should be an instance - but I do want this to be configurable. Meaning I want anyone to be able to pass any http adapter into this and not just be stuck with axios if that makes sense. So if I wanted to overwrite the axios one or I wanted to use something else different, I can make an adapter for that service and pass it to config. The idea using `new http` was that it was adhering to that contract which may not work ts

Reply:
I noticed a few areas in the code which seemed to mimic Axios, which then got me confused when they weren't really related. I would make the interface and then plug axios in. Rather than trying to overwrite axios.

Reply:
I hear you. I think your approach may be a better one. All I wanted was to be able to decoupe Axios from the app if possible. I want to allow injection of a different api that follows an interface, etc.